test/src/org/labkey/test/tests/targetedms/TargetedMSInstrumentNicknameTest.java

```diff
@@ -0,0 +1,258 @@
+        // We should see both model/serial numbers on the same page
+        assertTextPresent(Q_EXACTIVE, 4);  // Twice on the page itself, twice in hidden form elements
+        assertTextPresent(Q_EXACTIVE_SERIAL_ONLY, 2); // Once on the page itself, once in hidden form elements
+        assertTextPresent(
+                getProjectName() + "/" + QC_SUB_FOLDER + "/" + QC_SUB_SUB_FOLDER,
+                getProjectName() + "/" + NON_QC_SUB_FOLDER,
+
+                // Check for a sample without a serial number
+                REPLICATE_NAME_WITHOUT_SERIAL, FILE_PATH_WITHOUT_SERIAL,
+
+                // And one with a serial number
+                REPLICATE_NAME_WITH_SERIAL, FILE_PATH_WITH_SERIAL
+        );
+
+        // Rename the nickname for the one with the serial number to make sure they split
+        setFormElement(Locator.input("name"), NICKNAME_2);
+        clickButton("Save");
+
+        assertTextPresent(Q_EXACTIVE_SERIAL_ONLY, 2);  // Once visible on the page, once in a hidden form element
+        assertTextPresent(REPLICATE_NAME_WITH_SERIAL, FILE_PATH_WITH_SERIAL);
+
+        String postImpersonationUrl = getDriver().getCurrentUrl();
+        impersonateRole("Reader");
+        assertTextPresent(Q_EXACTIVE_SERIAL_ONLY, 1);  // Just the visible element, no form and hidden inputs for readers
+        stopImpersonating();
+        beginAt(postImpersonationUrl);
+
+        // Now resave, scoped to the folder
+        Locator.XPathLocator targetContainerLocator = Locator.name("targetContainerId");
+        selectOptionByTextContaining(targetContainerLocator.findElement(getDriver()), "In this folder");
+        clickButton("Save");
+
+        goToDashboard();
+        // We should see the current folder showing both nicknames
+        waitForElement(nickname1LinkLocator);
+        waitForElement(nickname2LinkLocator);
+        // We should also see the subfolder showing the nickname for one and the serial number for the other
+        waitForElement(qExactiveWithSerialLinkLocator);
+
+        // The subfolder should only see one of the nicknames
+        clickFolder(QC_SUB_SUB_FOLDER);
+        waitForElement(nickname1LinkLocator);
+        assertElementPresent(qExactiveWithSerialLinkLocator);
+        assertElementNotPresent(nickname2LinkLocator);
+
+        // Now try a non-QC folder, which should be inheriting the first nickname but not the second
+        clickFolder(NON_QC_SUB_FOLDER);
+        click(Locator.linkWithText(ISOTOPOLOGUE_FILE_ANNOTATED));
+        click(Locator.linkWithText("6 replicates"));
+        assertElementPresent(nickname1LinkLocator);
+        assertElementPresent(qExactiveWithSerialLinkLocator);
+        // We should also get links to the QC folders with data from the same instrument
+        assertElementPresent(Locator.linkWithText(getProjectName()));
+        assertElementPresent(Locator.linkWithText(QC_SUB_FOLDER));
+        assertElementPresent(Locator.linkWithText(QC_SUB_FOLDER));
```

Comment:
Should one of these links be to the subsub folder?
```suggestion
        assertElementPresent(Locator.linkWithText(QC_SUB_FOLDER));
        assertElementPresent(Locator.linkWithText(QC_SUB_SUB_FOLDER));
```